### PR TITLE
feat: support basic auth apis

### PR DESCRIPTION
**Pull request by MKMonkeyCat**
# Description

support basic auth apis

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -96,5 +96,16 @@ module.exports = {
 
     // allow return void
     'no-useless-return': 'off',
+
+    // import/no-extraneous-dependencies
+    'import/no-extraneous-dependencies': 'off',
```
Comment (SiongSng) [inline .eslintrc.js]: 請不要隨意修改 eslint 的設定

File: .vscode/settings.json [hunk 2]
```diff
@@ -13,5 +13,6 @@
   "i18n-ally.enabledParsers": ["json"],
   "i18n-ally.sourceLanguage": "en-US",
   "i18n-ally.displayLanguage": "zh-TW",
-  "i18n-ally.enabledFrameworks": ["vue"]
+  "i18n-ally.enabledFrameworks": ["vue"],
+  "files.eol": "\n"
```
Comment (SiongSng) [inline .vscode/settings.json]: what is eol?

**Review by SiongSng — CHANGES_REQUESTED** [review]

File: .eslintrc.js (hunk 1 above)
Comment (qmkcat) [reply .eslintrc.js]: image
ㄊ他會和 vue 的 path 衝突

Comment (SiongSng) [reply .eslintrc.js]: 那你把那個放在獨立的檔案裡面，不要全域停用

File: .editorconfig [hunk 3]
```diff
@@ -5,6 +5,7 @@ root = true
 charset = utf-8
 indent_style = space # indent style / 縮排格式 ( tab | space )
 indent_size = 2 # indent size / 缩排大小
+end_of_line = lf # end of line / 設定換行類型 ( lf | cr | crlf )
```
Comment (SiongSng) [inline .editorconfig]:
```suggestion

```

File: .eslintrc.js [hunk 4]
```diff
@@ -96,5 +96,13 @@ module.exports = {
 
     // allow return void
     'no-useless-return': 'off',
+
+    // no-param-reassign
```
Comment (SiongSng) [inline .eslintrc.js]: Please do not modify ESLint config.

File: src/configs/http.ts [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  /** The HTTP method used is not valid for the specified location | 使用的 HTTP 方法對指定的位置無效 */
+  METHOD_NOT_ALLOWED = 405,
+  /** Rate limited | 受到速率限制 */
+  TOO_MANY_REQUESTS = 429,
+  /** Server error | 伺服器錯誤 */
+  SERVER_ERROR = 500,
+}
+
+export const httpConfig = {
+  /** request headers | 請求標頭 */
```
Comment (SiongSng) [inline src/configs/http.ts]: Do not use Chinese comments in the code

File: src/configs/index.ts [hunk 6]
```diff
@@ -0,0 +1,15 @@
+import httpConfig from './http';
+
+export interface configsType {
+  /** There are settings for requests | 有關於請求的設定 */
+  http: typeof httpConfig;
+  /** send debug to server | 發送錯誤報告至伺服器 */
+  debugServer: boolean;
```
Comment (SiongSng) [inline src/configs/index.ts]: I don't think we need to send error reports to the server, right?

**Review by SiongSng — CHANGES_REQUESTED** [review]

**Comment by SiongSng** [comment]
[前端串連帳號系統 API](https://trello.com/c/sE7IVivc/1-%E5%89%8D%E7%AB%AF%E4%B8%B2%E9%80%A3%E5%B8%B3%E8%99%9F%E7%B3%BB%E7%B5%B1-api)

File: src/components/header/UserInfoHeader.vue [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+import { ref } from 'vue';
+
+import md5 from 'md5';
+import { useUserStore } from '@/stores/models/user';
+import { UserInfo } from '@/api/user/type';
+import PageLinksVue from '@/components/header/PageLinks.vue';
+
+const props = defineProps<{ info: UserInfo }>();
+
+const menuState = ref<undefined | ''>(undefined);
```
Comment (SiongSng) [inline src/components/header/UserInfoHeader.vue]: 應該有更好處理 menu 的作法?
例如用 boolean 搭配 css

File: src/http/index.ts [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+
+    config.headers = {
+      ...config.headers,
+      ...this.config.headers,
+    };
+
+    const store = useUserStore();
+    const token = store.token || config.token || this.config.token;
+    if (config.authentication && token) {
+      config.token = token;
```
Comment (SiongSng) [inline src/http/index.ts]:
```suggestion
```
直接從 store/config 取 token 就好，不必再設定一次

**Review by SiongSng — CHANGES_REQUESTED** [review]

**Review by SiongSng — APPROVED** [review]
LGTM

**Review by banahaker — APPROVED** [review]
Nice PR

**Pull request merged**